docker-compose.yml

```diff
@@ -1,5 +1,25 @@
 version: "3.9"
 services:
+  elasticsearch7:
+    image: elasticsearch:7.16.2
+    ports:
+      - "9200:9200"
+    environment:
+      - discovery.type=single-node
+      - bootstrap.memory_lock=true
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+    mem_limit: 1g
```

Comment:
Does the `mem_limit` setting override the `ES_JAVA_OPTS` setting, which looks like it may be "mem limit" related as well?

Reply:
From what I understand, the amount of memory given to the ES_JAVA_OPTS for elasticsearch should be half of the memory limit given to docker compose with mem_limit. I don't remember there being any explanation as for _why_ it should be half though, but since that did solve the out of memory docker errors, I just stuck with it.